src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -20,9 +29,60 @@
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Classification
 {
-    internal static class SemanticClassificationUtilities
+    internal partial class SemanticClassifier
```

Comment:
General architecture question: rather than putting this logic in to the classification code, should this be added into the async tagger itself? Each derived tagger would have to provide the serialization/deserialization functions, but then be able to apply it generally. I can understand the hesitation but I'm not sure it feels better getting jammed into here...